backend/sXb-integration-tests/Integration/ListingsController.cs

```diff
@@ -147,5 +147,321 @@ public class ListingsController : IClassFixture<WebApplicationFactory<Startup>>
+                        }
+                    }
+                    catch (KeyNotFoundException ex)
+                    {
+                        authorDict.Add(author, 1);
+                    }
+                }
+            }
+            bool sameAuthor = false;
+            int total;
+            if (content.TotalRecords > content.PageSize)
+            {
+                total = content.PageSize;
+            }
+            else
+            {
+                total = content.TotalRecords;
+            }
+            for (int i = 0; i < authorDict.Count; i++)
+            {
+                // If the authors reccurence is the same as total records count,
+                // that means, that at least one author reccurs in all records.
+                if (authorDict.ElementAt(i).Value == total)
+                {
+                    sameAuthor = true;
+                    break;
+                }
+            }
+
+            if (content.TotalRecords > 0)
+                Assert.True(sameAuthor);
+            else
+                Assert.False(sameAuthor);
+        }
+        [Theory]
+        [InlineData("9781976530739")]
+        [InlineData("harry+potter")]
+        [InlineData("Joanne+K+Rowling")]
+        public async Task Search_TermIs_X__Returns200WithPagingWithDataContaining_X_InAuthorOrTitleOrISBN(string query)
+        {
+            string url = $"/api/listings/search/{query}/1";
+            var client = _factory.CreateClient();
+
+            var response = await client.GetAsync(url);
+
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            Paging<ListingPreviewViewModel> content = await response.Content.ReadAsAsync<Paging<ListingPreviewViewModel>>();
+
+            // Log count of authors reccurence.
+            Dictionary<string, int> authorDict = new Dictionary<string, int>();
+
+            foreach (ListingPreviewViewModel vm in content.Data)
+            {
+                foreach (string author in vm.Authors)
+                {
+                    // if it's a new author initiate with the count of 1
+                    int countOfAuthor = 0;
+                    try
+                    {
+                        countOfAuthor = ++authorDict[author];
+                        if (authorDict[author] > 0)
+                        {
+                            authorDict[author] = countOfAuthor;
+                        }
+                    }
+                    catch (KeyNotFoundException ex)
+                    {
+                        authorDict.Add(author, 1);
+                    }
+                }
+            }
+            bool sameAuthor = false;
+            int total;
+            if (content.TotalRecords > content.PageSize)
+            {
+                total = content.PageSize;
+            }
+            else
+            {
+                total = content.TotalRecords;
+            }
+            for (int i = 0; i < authorDict.Count; i++)
```

Comment:
I'd like to talk about these tests. I think we could reduce the amount of code going into to them making them more maintainable. This should be the first thing we talk about Monday.